src/main/kotlin/uk/gov/dwp/dataworks/services/UserValidationService.kt

```diff
@@ -0,0 +1,37 @@
+package uk.gov.dwp.dataworks.services
+
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Service
+import uk.gov.dwp.dataworks.aws.AwsCommunicator
+import uk.gov.dwp.dataworks.logging.DataworksLogger
+
+@Service
+class UserValidationService {
+    companion object {
+        val logger: DataworksLogger = DataworksLogger(LoggerFactory.getLogger(UserValidationService::class.java))
+    }
+
+    @Autowired
+    private lateinit var awsCommunicator: AwsCommunicator
+    @Autowired
+    private lateinit var jwtParsingService: JwtParsingService
+
+    fun checkAttributes(jwt: String): Boolean {
+        val jwtObject= jwtParsingService.parseToken(jwt)
+        return checkJwtForGroup(jwtObject.username, jwtObject.cognitoGroups) && checkForEnabledKms(jwtObject.username)
+    }
+
+    fun checkForEnabledKms(userName: String): Boolean {
+        return awsCommunicator.checkForExistingEnabledKey(userName)
+    }
+
+    fun checkJwtForGroup(userName: String, cognitoGroups: List<String>): Boolean {
```

Comment:
This function is incorrect. We are not checking if the username is also a cognito group?!?!  We are just checking for the existance of a group. i.e. the list is not empty